### ISSUE TITLE
package_id will be resued if a new package is created right after a deletion

### ISSUE THREAD

**Issue by gfxcc** [gfxcc/san11-platform#19]
This will cause confusing in many places as `package_id` is expected to be unique across all created pacakges.

An `activity` could reference a resource while the resource_id is resued for a newly created resource.

**Comment by gfxcc** [comment]
We should use uuid() instead of sequential integer as the resource_id (primary key) for following reasons
- current implementaion of getting the next sequential id will potentially reused id from deleted resource.
- sequential primary key will likely create hotspot in distributed DB. (many distributed database will partition table based on primary key. Have sequential primary will cause most new adding operation be redirected to a single host)
- sequential primary key expose the internal details and user may try to hack the system. E.g. if a pakcage url is `packages/12`, user may try to visit `packages/13`. We have seen such attempts before)
- use uuid() as primary key (resource id) is relative easier to implement. (currently, we have to run a SELECT to get the next sequential id)

**Comment by gfxcc** [comment]
This should be addressed along with Package data model migration.

**Comment by gfxcc** [comment]
> We should use uuid() instead of sequential integer as the resource_id (primary key) for following reasons
> 
> * current implementaion of getting the next sequential id will potentially reused id from deleted resource.
> * sequential primary key will likely create hotspot in distributed DB. (many distributed database will partition table based on primary key. Have sequential primary will cause most new adding operation be redirected to a single host)
> * sequential primary key expose the internal details and user may try to hack the system. E.g. if a pakcage url is `packages/12`, user may try to visit `packages/13`. We have seen such attempts before)
> * use uuid() as primary key (resource id) is relative easier to implement. (currently, we have to run a SELECT to get the next sequential id)

Moved forward with using imbeded incremental integer (serial in postgres) as `resource_id`. Current model which use `SELECT MAX(resource_id) FROM xxxx WHERE xxx` causing another issue which will run into id collision when multiple creation is running in parallel.

uuid is not used since
- want to avoid having long url like `packages/afd1b62f-5705-4037-97cb-9c7dbe28e412/binaries/88e08c7d-1439-43f1-bd89-47575f374acb`.
- easier to adopt as current resource_id is also an integer.

**Comment by gfxcc** [comment]
Moved forward with using sequential primary key. The limits from that approach is considered acceptable for now.

> sequential primary key expose the internal details and user may try to hack the system. E.g. if a pakcage url is packages/12, user may try to visit packages/13. We have seen such attempts before)
